lib/Reader/readerir.cpp

```diff
@@ -4000,6 +3999,21 @@ void GenIR::storePrimitiveType(IRNode *Value, IRNode *Addr,
   StoreInst->setAlignment(Align);
 }
 
+void GenIR::storeNonPrimitiveType(IRNode *Value, IRNode *Addr,
+                                  CORINFO_CLASS_HANDLE Class,
+                                  ReaderAlignType Alignment, bool IsVolatile,
+                                  CORINFO_RESOLVED_TOKEN *ResolvedToken,
+                                  bool IsField) {
+  // Get the minimum Alignment for the class
+  Alignment = getMinimumClassAlignment(Class, Alignment);
+  bool IsValueIsPointer = structsAreRepresentedByPointers();
+  if (Value->getType()->isVectorTy()) {
+    IsValueIsPointer = false;
+  }
+  rdrCallWriteBarrierHelper(Addr, Value, Alignment, IsVolatile, ResolvedToken,
```

Comment:
Since this code now lives in GenIR we should delete structsAreRepresentedByPointers method. I would change the code here to:
`bool IsValueIsPointer = !Value->getType()->isVectorTy();`
`assert(!IsValueIsPointer || doesValueRepresentStruct(Value));`